sdk/src/wallet/operations/transaction/input_selection.rs

```diff
@@ -72,173 +82,76 @@ where
         let available_outputs_signing_data = filter_inputs(
             &wallet_data,
             wallet_data.unspent_outputs.values(),
-            slot_index,
+            creation_slot_index,
             protocol_parameters.committable_age_range(),
-            custom_inputs.as_ref(),
-            mandatory_inputs.as_ref(),
+            &options.mandatory_inputs,
         )?;
 
         let mut mana_rewards = HashMap::new();
 
-        if let Some(burn) = burn {
+        if let Some(burn) = &options.burn {
             for delegation_id in burn.delegations() {
                 if let Some(output) = wallet_data.unspent_delegation_output(delegation_id) {
                     mana_rewards.insert(
                         output.output_id,
                         self.client()
-                            .get_output_mana_rewards(&output.output_id, slot_index)
+                            .get_output_mana_rewards(&output.output_id, creation_slot_index)
```

Comment:
Is `creation_slot_index` correct here? Shouldn't it be the slot index of the commitment input that will be referenced?

Reply:
maybe! I have no idea tbh

Reply:
To me it sounds so
```
If the output is in Delegating State let Delegation End be Future Bounded Epoch Index - 1, where:
Future Bounded Slot Index is given by Commitment Index + Min Committable Age where Commitment Index is the slot index of the commitment input. If no Commitment Input is present, the transaction is invalid.
Note that the presence of a Commitment Input is already required for any transaction containing a Delegation Output on the input or output side, except when it is destroyed in the Delayed Claiming State.
```
https://github.com/iotaledger/tips/blob/tip40/tips/TIP-0040/tip-0040.md#calculations-5

Reply:
I'm also not sure, opened an issue https://github.com/iotaledger/iota-sdk/issues/1995

Reply:
Philipp answered on the issue, we need the commitment slot index here

Reply:
can this be resolved?

Reply:
comment was hidden stupid github